src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestorer.cs

```diff
@@ -101,12 +101,13 @@ private ProcessArgumentBuilder GetArguments(string root, DotNetCoreRestoreSettin
                     builder.AppendQuoted(runtime);
                 }
             }
-
+#pragma warning disable 0618
             // Quiet
-            if (settings.Quiet)
+            if (settings.Quiet && !settings.Verbosity.HasValue)
             {
-                builder.Append("--quiet");
+                settings.Verbosity = DotNetCoreRestoreVerbosity.Minimal;
```

Comment:
I don't think we should set this propery automatically. Better to write to the log as a warning giving the same hint as the obsolete message.